packages/core/services/geocoder-service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@angular/core';
+import { bindCallback, from, Observable, of, throwError } from 'rxjs';
+import { flatMap, switchMap } from 'rxjs/operators';
+import { Geocoder, GeocoderRequest, GeocoderResult, GeocoderStatus } from './google-maps-types';
+import { MapsAPILoader } from './maps-api-loader/maps-api-loader';
+
+declare var google: any;
+
+@Injectable()
+export class AgmGeocoder {
+  protected readonly geocoder$: Observable<Geocoder>;
+
+  constructor(loader: MapsAPILoader) {
+    this.geocoder$ = from(loader.load()).pipe(
+      flatMap(() => this._createGeocoder())
```

Comment:
since geocoder object construction is synchronous, why not use `map` instead of `flatMap`, and have the `_createGeocoder` return the geocoder, and not an observable wrapper?

Reply:
To be honest with you, I've done it like that because I'm not very experienced with reactive programming. Now I can see what you mean though and will fix appropriately.